src/machine.rs

```diff
@@ -259,6 +259,20 @@ impl<'m> Machine<'m> {
         self.pid
     }
 
+    /// Check if the jailer/firecracker process is running
+    pub fn is_running(&self) -> bool {
```

Comment:
Actually I don't think this function should be needed in `firec`. If the VMM is running, we should have its PID and not if it's not running. If the firecracker process dies of external causes (which should be considered abnormal), we find out next time we interact with it. We might want to look into monitoring the process actively later but that'd involve an ask task or thread to be launched by fc, with the latter being an undesirable option (people generally don't like threads being launched by libraries behind their back as its unexpected).

Reply:
Also the implementation is quite heavy and does I/O and used from async functions so we need to make it async, if we do keep it after all.

Reply:
good points